clang/lib/Driver/Driver.cpp

```diff
@@ -4041,6 +4017,10 @@ class ToolSelector final {
     if (!T)
       return nullptr;
 
+    // Classic Flang is not integrated with the backend.
+    if (C.getDriver().IsFlangMode() && !T->hasIntegratedAssembler())
+      return nullptr;
+
```

Comment:
Need an `#ifdef ENABLE_CLASSIC_FLANG` ?

Reply:
`T->hasIntegratedAssembler` returns `true` for the new Flang, so this works without a macro, which I thought was preferable. It would be great if this could be upstreamed into llvm-project.